src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.cs

```diff
@@ -173,26 +179,28 @@ public static bool TryParse(ReadOnlySpan<byte> source, out int value, out int by
         /// </exceptions>
         public static bool TryParse(ReadOnlySpan<byte> source, out long value, out int bytesConsumed, char standardFormat = default)
         {
-            switch (standardFormat)
+        FastPath:
+            if (standardFormat == default)
+            {
+                return TryParseInt64D(source, out value, out bytesConsumed);
+            }
+
+            switch (standardFormat | 0x20) // convert to lowercase
             {
-                case default(char):
                 case 'g':
-                case 'G':
                 case 'd':
-                case 'D':
-                    return TryParseInt64D(source, out value, out bytesConsumed);
+                    standardFormat = default;
+                    goto FastPath;
```

Comment:
Why not move the label inside the `if` and avoid the comparison, since `standardFormat` will obviously be `default`?

Reply:
Illegal syntax in C#. The JIT is smart enough to optimize the check away anyway and jmp directly within the block.